database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/PostsView.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Copyright (c) 2021 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import Firebase
+
+struct PostsView: View {
+  @StateObject var user = UserViewModel()
+  var title: String
+  var tabOpened: UserViewModel.Tab
```

Comment:
I would like to complain on the `tabOpened` name again 😄. I think we should:
1. Move the enum to either top level or under PostViewModel
2. Rename it to something like `PostListType`
3. Rename the variable here to something like `postsType`

I'm not too strong about it, also it's not a blocker for the current PR.

Reply:
I am not too sure how I can implement the enum in top level or under `PostViewModel` so that when I call a function in `UserViewModel` it is still in scope.

Reply:
I would suggest the following. The enum looks independent enough to be placed in a separate file. If you just place it into a separate file, it will be in the top level app module scope, so it will be available at anywhere within your app module which is what you need.